base/util.jl

```diff
@@ -795,6 +795,15 @@ function _kwdef!(blk, params_args, call_args)
     blk
 end
 
+"""
+    Base.assume(cond::Bool)
+
+Assume that a condition `cond` always hold. This information may be used by the compiler for
```

Comment:
I initially confused this with `expect`/`likely`. Maybe use:

> Asserts that a condition `cond` always holds. This assumption may be used by the compiler for
optimization purposes. 

So that the definition is not cycle (e.g. defining assume by using the common word assume) and add a sentence about what happens if the information is incorrect.